### PR TITLE
KEP-1880 Multiple Service CIDRs: Graduate to Beta (2/2)

### PR DESCRIPTION
**Pull request by aojea**
/kind feature

Update the ServiceCIDR feature gate and APIs to beta

KEP https://github.com/kubernetes/enhancements/issues/1880

```release-note
Update the feature MultiCIDRServiceAllocator to beta (disabled by default). Users need to enable the feature gate and the networking v1beta1 group to be able to use this new feature, that allows to dynamically reconfigure Service CIDR ranges.
```

```docs
- [KEP]: https://github.com/kubernetes/enhancements/issues/1880
```

### CODE REVIEW

File: pkg/registry/core/service/ipallocator/metrics.go [hunk 1]
```diff
@@ -73,6 +74,17 @@ var (
 		},
 		[]string{"cidr", "scope"},
 	)
+	clusterIPAllocationLatency = metrics.NewHistogramVec(
```
Comment (dashpole) [inline pkg/registry/core/service/ipallocator/metrics.go:77]: @dgrisonnet to review new metric

Comment (dgrisonnet) [reply pkg/registry/core/service/ipallocator/metrics.go:77]: I commented on 1/2 and the new metric looks good to me

File: pkg/kubeapiserver/default_storage_factory_builder.go [hunk 2]
```diff
@@ -71,8 +71,8 @@ func NewStorageFactoryConfig() *StorageFactoryConfig {
 		//
 		// TODO (https://github.com/kubernetes/kubernetes/issues/108451): remove the override in 1.25.
 		// apisstorage.Resource("csistoragecapacities").WithVersion("v1beta1"),
-		networking.Resource("ipaddresses").WithVersion("v1alpha1"),
-		networking.Resource("servicecidrs").WithVersion("v1alpha1"),
+		networking.Resource("ipaddresses").WithVersion("v1beta1"),
+		networking.Resource("servicecidrs").WithVersion("v1beta1"),
```
Comment (aojea) [inline pkg/kubeapiserver/default_storage_factory_builder.go:75]: @liggitt is this correct? I'm always confused with this part

**Comment by aojea** [comment]
/assign @thockin 

mechanical changes for moving to beta

**Comment by thockin** [comment]
Thanks!

/lgtm
/approve

**Pull request merged**